zetaclient/chains/solana/observer/inbound.go

```diff
@@ -281,7 +290,39 @@ func (ob *Observer) BuildInboundVoteMsgFromEvent(event *clienttypes.InboundEvent
 		0,
 		event.CoinType,
 		event.Asset,
-		ob.ZetacoreClient().GetKeys().GetOperatorAddress().String(),
 		0, // not a smart contract call
+		crosschaintypes.ProtocolContractVersion_V1,
+		false, // not relevant for v1
 	)
+
+	// make sure the message is valid before posting to zetacore
+	err = msg.ValidateBasic()
+	if err != nil {
+		ob.Logger().Inbound.Error().Err(err).Fields(lf).Msg("invalid inbound vote message")
+		return nil
+	}
+
+	return msg
+}
+
+// CheckEventProcessability checks if the inbound event is processable
+func (ob *Observer) CheckEventProcessability(event clienttypes.InboundEvent) bool {
```

Comment:
If the function returns a bool it should be named `IsX`
`IsEventProcessable`

Reply:
renamed across all the chains: https://github.com/zeta-chain/node/pull/3184/commits/5e4e6d7ef2ef2112c8d7bae7ddf1976eccd684aa